daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -293,9 +297,9 @@ downloadAllReachablePackages args pid = loop Map.empty [pid]
         then loop acc morePids
         else do
           pkg <- downloadPackage args pid
-          loop (Map.insert pid pkg acc) (packageRefs pkg ++ morePids)
+          loop (Map.insert pid pkg acc) (nubSort $ packageRefs pkg ++ morePids)
```

Comment:
can we use a Set instead of `nubSort`ing a list? I think it might also be benefitial to only insert `packageRefs pkg` that aren’t already in the map instead of first inserting them here only to then drop them in the next step.

Reply:
I added a filter for the `packageRefs pkg` so that we exclude package ids that already in the accumulator. I think we should keep the list and `nubSort` because we loop over the list. Otherwise, you'll end up with a lot of `toList/fromList` in the iteration.

Reply:
You can use `minView` https://hackage.haskell.org/package/containers-0.6.4.1/docs/Data-Set.html#v:minView but fair enough, I don’t feel strongly about this at all.

Reply:
ah cool, I didn't know `minView`. Changed it to use this.